doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,491 @@
++     |           +--ro oc-sys-ext:selMode?   string
++     |           +--ro oc-sys-ext:refid?     inet:host
++     |           +--ro oc-sys-ext:type?      string
++     |           +--ro oc-sys-ext:now?       uint32
++     |           +--ro oc-sys-ext:reach?     uint8
+
+
+module: sonic-system-ntp
+
++    +--rw sonic-system-ntp
++       +--rw NTP
++       |  +--rw NTP_LIST* [global_key]
++       |     +--rw global_key    enumeration
++       |     +--rw source?       union
++       |     +--rw vrf?          string
++       +--rw NTP_SERVER
++          +--rw NTP_SERVER_LIST* [server_address]
++             +--rw server_address    inet:host
+
+```
+
+### 3.6.2 CLI
+
+
+#### 3.6.2.1 Configuration Commands
+All commands are executed in `configuration-view`:
+```
+sonic# configure terminal
+sonic(config)#
+```
+
+##### 3.6.2.1.1 Configure NTP server
+```
+sonic(config)#ntp
+    server              Configure NTP server
+sonic(config)#ntp server 
+String  NTP server address or name
+
+sonic(config)# ntp server 10.11.0.1
+
+sonic(config)# ntp server 2001:aa:aa::a
+
+sonic(config)# ntp server pool.ntp.org
+
+```
+
+##### 3.6.2.1.2 Delete NTP server
+
+```
+sonic(config)# no ntp server
+  String  NTP server address or name
+
+sonic(config)# no ntp server 10.11.0.1
+
+sonic(config)# no ntp server 2001:aa:aa::a
+
+sonic(config)# no ntp server pool.ntp.org
+
+```
+
+##### 3.6.2.1.3 Configure NTP source ip
+
+```
+sonic(config)# ntp
+    source              Configure source IP address
+
+sonic(config)# ntp source 11.22.33.55
+
+sonic(config)# ntp source 2001:aa:aa::b
+
+```
+
+##### 3.6.2.1.4 Delete NTP source
+
+```
+sonic(config)# no ntp source
+
+```
+
+##### 3.6.2.1.5 Configure NTP vrf
+
+```
+sonic(config)#
+    vrf          Enabling NTP on a VRF
+
+sonic(config)#ntp vrf
+  management                  Enable NTP on management VRF
+  String(Max: 32 characters)  Enable NTP on non-default VRF
+
+sonic(config)# ntp vrf management
+
+```
+
+##### 3.6.2.1.6 Delete NTP vrf
+
+```
+sonic(config)# no ntp
+    vrf  Disable NTP on a VRF
+
+sonic(config)# no ntp vrf
+  management                  Disable NTP on management VRF
+  String(Max: 32 characters)  Disable NTP on non-default VRF
+
+sonic(config)# no ntp vrf management
+
+```
+
+#### 3.6.2.2 Show ntp associations
+
+```
+sonic(config)# do show ntp
+  associations  NTP associations
+
+
+sonic(config)# do show ntp associations
+     remote           refid      st t when poll reach   delay   offset  jitter
+==============================================================================
+*10.11.0.1       10.11.8.1        4 u   28   64    1    0.183    1.499   2.625
++2001:aa:aa::b   60.39.129.68    10 u   27   64    1    0.638  2171.31   0.411
++10.11.0.2       10.11.8.1        4 u   24   64    1    0.240  -13.957  12.786
+* master (synced), # master (unsynced), + selected, - candidate, ~ configured
+
+```
+
+#### 3.6.2.3 Debug Commands
+
+#### 3.6.2.4 IS-CLI Compliance
+

```

Comment:
Please fill this out - we should be drop-in replace to the extent possible.

Reply:
sure, will add.